src/main/java/org/broadinstitute/consent/http/db/mapper/InstitutionMapper.java

```diff
@@ -0,0 +1,37 @@
+package org.broadinstitute.consent.http.db.mapper;
+
+import org.broadinstitute.consent.http.models.Institution;
+import org.jdbi.v3.core.mapper.RowMapper;
+import org.jdbi.v3.core.statement.StatementContext;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.HashMap;
+import java.util.Map;
+
+public class InstitutionMapper implements RowMapper<Institution>, RowMapperHelper {
+
+  private final Map<Integer, Institution>  institutionMap = new HashMap<>();
+
+
+  @Override
+  public Institution map(ResultSet resultSet, StatementContext statementContext) throws SQLException {
+    Institution institution;
+
+    if (institutionMap.containsKey(resultSet.getInt("institution_id"))) {
+      institution = institutionMap.get(resultSet.getInt("institution_id"));
+    } else {
+      institution = new Institution();
+      institution.setId(resultSet.getInt("institution_id"));
+    }
```

Comment:
You could pull the `institution_id` call out to an int above the if clause. That would clean up the copy/past of `resultSet.getInt("institution_id"))`
```suggestion
    int institutionId = resultSet.getInt("institution_id"));
    if (institutionMap.containsKey(institutionId) {
      institution = institutionMap.get(institutionId);
    } else {
      institution = new Institution();
      institution.setId(institutionId);
    }
```